core/src/main/java/hudson/model/Fingerprint.java

```diff
@@ -1348,83 +1313,64 @@ public Api getApi() {
     }
 
     /**
-     * The file we save our configuration.
+     * Loads a {@link Fingerprint} from the Storage with the given unique id.
+     * @return Loaded {@link Fingerprint}. Null if the config file does not exist or
+     * malformed.
      */
-    private static @NonNull XmlFile getConfigFile(@NonNull File file) {
-        return new XmlFile(XSTREAM,file);
+    public static @CheckForNull Fingerprint load(@NonNull String id) throws IOException {
+        long start=0;
+        if(logger.isLoggable(Level.FINE)) {
+            start = System.currentTimeMillis();
+        }
+
+        Fingerprint loaded = FingerprintStorage.get().load(id);
+        initFacets(loaded);
+
+        if(logger.isLoggable(Level.FINE)) {
+            logger.fine("Loading fingerprint took " + (System.currentTimeMillis() - start) + "ms");
+        }
+
+        return loaded;
     }
 
     /**
-     * Determines the file name from md5sum.
+     * @deprecated as of TODO
```

Comment:
Same as above, guidelines are needed. I would also keep the existing API documentation even if it is deprecated

```suggestion
     * Determines the file name from md5sum.
     * @deprecated as of TODO
```